src/mlpack/methods/ann/layer/mean_pooling.hpp

```diff
@@ -160,12 +160,21 @@ class MeanPooling
   template<typename eT>
   void Pooling(const arma::Mat<eT>& input, arma::Mat<eT>& output)
   {
+    arma::Mat<eT> inputPre = input;
```

Comment:
Could think about a workaround to remove the extra copy here, I guess if we want to keep the input unmodified there is no way around it.

Reply:
Yes you are right.